geps/gep-1897.md

```diff
@@ -125,7 +126,235 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+    // backend pod.
+    //
+    // A single TrustedCACertRef to a Kubernetes Secret has "Core"
+    // support.  Implementations MAY choose to support attaching
+    // multiple certificates to a backend, but this behavior is 
+    // implementation-specific.
+    //
+    // References to a resource in a different namespace are 
+    // invalid.
+    // 
+    // This field is required for any TLSConnectionPolicyConfig.
+    // 
+    // Support: Core - A single reference to a Kubernetes Secret of type kubernetes.io/tls
+    //
+    // Support: Implementation-specific (More than one reference or other resource types)
+    //
+    // +kubebuilder:validation:MaxItems=64
+    // +kubebuilder:validation:MinItems=1
+    TrustedCACertRefs []SecretObjectReference `json:”trustedCACertRefs”`
+
+    // Port is the network port that the implementation watches to
+    // know if the connection should be TLS and the targetRef’s
+    // certificate should be validated by the certs in TrustedCACertRefs
+    // If empty, then all ports for the targetRef are watched to know
+    // if the connection should be TLS, the targetRef’s certificate
+    // should be validated by the certs in TrustedCACertRefs, and a
+    // status delivered in the response for validation failures.
+    Port PortNumber `json:port,omitempty`
+}
+
+// TLSConnectionPolicyStatus defines the observed state of TLSConnectionPolicy.
+type TLSConnectionPolicyStatus struct {
+    // Conditions describe the current conditions of the TLSConnectionPolicy.
+    //
+    // Implementations should prefer to express TLSConnectionPolicy
+    // conditions using the `TLSConnectionPolicyConditionType` and
+    // `TLSConnectionPolicyConditionReason` constants so that 
+    // operators and tools can converge on a common vocabulary to 
+    // describe TLSConnectionPolicy state.
+    // Known condition types are:
+    // 
+    // * “Accepted”
+    // 
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:default={type: "Accepted", status: "Unknown", reason:"Pending", message:"Waiting for validation", lastTransitionTime: "1970-01-01T00:00:00Z"}
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// TLSConnectionPolicyConditionType is the type of a condition used
+// as a signal by TLSConnectionPolicy.  This type should be used with
+// the TLSConnectionPolicyStatus.Conditions field.
+type TLSConnectionPolicyConditionType string
+
+//  TLSConnectionPolicyConditionReason is a reason that explains why a
+// particular TLSConnectionPolicyConditionType was generated.
+type TLSConnectionPolicyConditionReason string
+
+const (
+    // This condition indicates that the TLSConnectionPolicy has been
+    // accepted as valid.
+    // Possible reason for this condition to be True is:
+    //
+    // * “Accepted” 
+    // 
+    // Possible reasons for this condition to be False are:
```

Comment:
can `ResolvedRefs` condition be also used here? since the policy addresses external references - service, secrets

Reply:
Sounds reasonable, unless there are privacy issues.  As an implementor, what kind of condition reasons would you think should be there?  Should there be a general validation on the certificate so it could have an invalid cert reason?  Or does this store results of an analysis of reference grants between the policy and targets?

Reply:
I think _most_ implementations would only look at this if they were implementing a Service that was targeted by one of these policies, so I don't think `ResolvedRefs` would be relevant in that case, but I agree that it likely would be very useful for the secret reference(s) that will be included.

Reply:
This one I will pass on unless there is further interest.